helix-term/src/commands.rs

```diff
@@ -2926,6 +2927,42 @@ fn buffer_picker(cx: &mut Context) {
     cx.push_layer(Box::new(overlaid(picker)));
 }
 
+fn command_history_picker(cx: &mut Context) {
+    let command_history = cx.editor.registers.read(':', cx.editor);
+
+    let items: Vec<Command> = command_history
```

Comment:
Perhaps use a set? A [`BTreeSet`](https://doc.rust-lang.org/stable/std/collections/struct.BTreeSet.html) to rather than HashSet to maintain ordering

Reply:
I think a BTreeSet wouldn't help, b/c we need insertion order, not lexicographical order.